compiler/atomic.go

```diff
@@ -16,6 +17,20 @@ func (b *builder) createAtomicOp(call *ssa.CallCommon) (llvm.Value, bool) {
 	case "AddInt32", "AddInt64", "AddUint32", "AddUint64", "AddUintptr":
 		ptr := b.getValue(call.Args[0])
 		val := b.getValue(call.Args[1])
+		if strings.HasPrefix(b.Triple, "avr") {
+			// AtomicRMW does not work on AVR as intended:
+			// - There are some register allocation issues (fixed by https://reviews.llvm.org/D97127 which is not yet in a usable LLVM release)
```

Comment:
This is part of LLVM 13, so that particular issue shouldn't be a concern anymore.